roles/puppet_repositories/tasks/debian.yml

```
@@ -5,6 +5,12 @@
     state: present
     update_cache: true
 
+- name: Remove any existing Puppet release packages
+  package:
+    name: "puppet{{ item }}-release"
+    state: absent
+  loop: "{{ puppet_repositories_supported_versions | difference([puppet_repositories_version]) }}"
```

```suggestion
  loop: "{{ puppet_repositories_supported_versions | map('string') | difference([puppet_repositories_version | string]) }}"
```

this should fix the issue I was talking about